libs/trpc-openapi/src/generator/schema.ts

```diff
@@ -12,7 +12,18 @@ import {
 const zodSchemaToOpenApiSchemaObject = (
   zodSchema: z.ZodType
 ): OpenAPIV3.SchemaObject => {
-  return zodToJsonSchema(zodSchema, { target: 'openApi3' })
+  /**
+   * MODTREE WAS HERE
+   *
+   * disable refStrategy because trpc-openapi does not support
+   * recursive types in docs due to its implementation
+   *
+   * trpc-openapi#37
+   */
+  return zodToJsonSchema(zodSchema, {
+    target: 'openApi3',
+    $refStrategy: 'none',
+  })
```

Comment:
`trpc-openapi` implementation means the generated docs errors on recursive types like `prereqTree`.

Option 1: Fix implementation to allow YAML refs
Option 2: Disable YAML refs

Both options give essentially the same end result. I chose option 2 because it only needs a small change to `trpc-openapi`.